src/integration/java/io/pinecone/integration/dataPlane/serverless/QueryErrorTest.java

```diff
@@ -0,0 +1,116 @@
+package io.pinecone.integration.dataPlane.serverless;
+
+import io.grpc.StatusRuntimeException;
+import io.pinecone.clients.AsyncIndex;
+import io.pinecone.clients.Index;
+import io.pinecone.clients.Pinecone;
+import io.pinecone.exceptions.PineconeValidationException;
+import io.pinecone.helpers.RandomStringBuilder;
+import io.pinecone.proto.DescribeIndexStatsResponse;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.openapitools.client.model.IndexModelSpec;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import static io.pinecone.helpers.BuildUpsertRequest.generateVectorValuesByDimension;
+import static io.pinecone.helpers.IndexManager.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class QueryErrorTest {
+
+    private static Index index;
+    private static AsyncIndex asyncIndex;
+    private static final int dimension = 3;
+
+    @BeforeAll
+    public static void setUp() throws InterruptedException {
+        String apiKey = System.getenv("PINECONE_API_KEY");
+        String indexType = IndexModelSpec.SERIALIZED_NAME_SERVERLESS;
+        Pinecone pinecone = new Pinecone(apiKey);
+
+        String indexName = findIndexWithDimensionAndType(pinecone, dimension, indexType);
+        if (indexName.isEmpty()) indexName = createNewIndex(pinecone, dimension, indexType);
+        index = pinecone.getIndex(indexName);
+        asyncIndex = pinecone.getAsyncIndex(indexName);
+    }
+
+    @Test
+    public void queryWithIncorrectVectorDimensionSync() {
+        String namespace = RandomStringBuilder.build("ns", 8);
+
+        DescribeIndexStatsResponse describeIndexStatsResponse1 = index.describeIndexStats(null);
```

Comment:
The instance index created using getIndex() is used here for data plane call describeIndexStats. More tests for getIndex() and getAsyncIndex() will be added in serverless tests ticket.